modules/library/main/src/main/java/org/geotools/styling/SLDTransformer.java

```diff
@@ -375,22 +368,20 @@ public void visit(Stroke stroke) {
             encodeCssParam("stroke-width", stroke.getWidth(), 1.0);
             encodeCssParam("stroke-dashoffset", stroke.getDashOffset(), 0.0);
 
-            float[] dash = stroke.getDashArray();
-
-            if (dash != null) {
-                StringBuffer sb = new StringBuffer();
+            encodeStrokeDasharray(stroke.dashArray());
 
-                for (int i = 0; i < dash.length; i++) {
-                    sb.append(dash[i]);
-                    if (i < dash.length - 1) {
-                        sb.append(" ");
-                    }
-                }
-
-                encodeCssParam("stroke-dasharray", ff.literal(sb.toString()));
+            end("Stroke");
+        }
 
+        private void encodeStrokeDasharray( List<Expression> expressions) {
+            if (expressions == null) return;
+            AttributesImpl attributes = new AttributesImpl();
+            attributes.addAttribute("", "name", "name", "", "stroke-dasharray");
+            start("CssParameter", attributes);
+            for (Expression expression : expressions) {
```

Comment:
This one seems to be problematic. If one starts with a SLD standard dasharray as text, like "10 10" the encoder should return the same, and roll the mixed mode expressions only if the expressions are not literals.
The idea is that if we get a OGC compliant SLD in, it should be compliant when we rewrite it, too